bottlerocket/agents/src/sonobuoy.rs

```diff
@@ -143,72 +144,102 @@ pub fn results_sonobuoy(
         serde_json::from_str(&stdout).context(error::DeserializeJsonSnafu)?;
     trace!("The sonobuoy results are valid json");
 
-    let e2e_status = run_status
-        .get("plugins")
-        .context(error::MissingSonobuoyStatusFieldSnafu { field: "plugins" })?
-        .as_array()
-        .context(error::MissingSonobuoyStatusFieldSnafu { field: "plugins" })?
-        .first()
-        .context(error::MissingSonobuoyStatusFieldSnafu {
-            field: format!("plugins.{}", sonobuoy_config.plugin),
-        })?;
-
-    // Sometimes a helpful log is available in the progress field, but not always.
-    let progress_status = e2e_status
-        .get("progress")
-        .map(|value| value.to_string())
-        .unwrap_or_else(|| "".to_string());
-
-    let result_status = e2e_status
-        .get("result-status")
-        .context(error::MissingSonobuoyStatusFieldSnafu {
-            field: format!("plugins.{}.result-status", sonobuoy_config.plugin),
-        })?
-        .as_str()
-        .context(error::MissingSonobuoyStatusFieldSnafu {
-            field: format!("plugins.{}.result-status", sonobuoy_config.plugin),
-        })?;
-
-    let result_counts = run_status
+    process_sonobuoy_test_results(&run_status)
+}
+
+/// process_sonobuoy_test_results parses the output from `sonobuoy status --json` output and gets
+/// the overall status of the plugin results.
+pub(crate) fn process_sonobuoy_test_results(
+    run_status: &serde_json::Value,
+) -> Result<TestResults, error::Error> {
+    let mut num_passed: u64 = 0;
+    let mut num_failed: u64 = 0;
+    let mut num_skipped: u64 = 0;
+    let mut progress = Vec::new();
+    let mut outcome_summary = HashMap::from([
+        ("pass", 0),
+        ("passed", 0),
+        ("fail", 0),
+        ("failed", 0),
+        ("timeout", 0),
+        ("timed-out", 0),
```

Comment:
In the theoretical struct you wouldn't need separate fields for the alternate spellings.

Reply:
I may need a clearer picture of what you're envisioning here. It looks like plugins can report either variation for each result, so somewhere we need to evaluate them.

My thought with this structure was we wouldn't have to check each value every time, just collect it all up and get the totals at the end.